### DISCUSSION TITLE
Are traits still a concept in MTP?

### DISCUSSION THREAD

**Discussion by thomhurst** [microsoft/testfx#3991] [Q&A] [answered]
I've had an issue raised, and just wondering if I need to do anything or if it's not a concept in the new platform.

https://github.com/thomhurst/TUnit/issues/1145

**Comment by thomhurst** [comment]
@Evangelink @nohwnd @MarcoRossignoli wondering if you could help please

**Comment by nohwnd** [comment]
It is a thing and should remain one, both the category (key, value) and trait (just value), this is how we handle them in vstest bridge:

https://github.com/microsoft/testfx/blob/d3abcd56cde37699f9d5259514c87d3907b4442e/src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ObjectModelConverters.cs#L97-L108 

to convert to SerializableNamedKeyValuePairsStringProperty https://grep.app/search?q=SerializableNamedKeyValuePairsStringProperty

**Reply by thomhurst** [reply]
Thanks, I'll give this a go later!

**Reply by thomhurst** [reply]
@nohwnd `SerializableNamedKeyValuePairsStringProperty` is internal. Does it need opening up? Or maybe a traits property makes more sense to abstract away the internals and needing to know a "traits" key for framework writers.

**Comment by Youssef1313** [comment, marked as answer]
I know it's quite old discussion and you likely figured out already. But for future readers, `TestMetadataProperty` is the thing to use.
